ext/RMagick/extconf.rb

```diff
@@ -186,6 +159,27 @@ def exit_failure(msg)
       exit(1)
     end
 
+    def determine_imagemagick_package
+      exit_failure "Can't install RMagick #{RMAGICK_VERS}. Can't find pkg-config in #{ENV['PATH']}\n" unless find_executable('pkg-config')
+
+      packages = `pkg-config --list-all`.scan(/(ImageMagick\-6[\.A-Z0-9]+) .*/)
+
+      if packages.length.zero?
+        exit_failure "Can't install RMagick #{RMAGICK_VERS}. Can't find ImageMagick with pkg-config\n"
+      end
+
+      if packages.length > 1
```

Comment:
I could almost see using a `case` statement for these:

```rb
case packages.length
when 1
  packages.first
when 0
  exit_failure...
else
  Logging.message...
end
```

Reply:
I personally think it is more clear with the if statements but I am happy to change this if you prefer to use a `case` instead.

Reply:
No worries, I'm fine leaving it as is. I think the fact that we need to return `packages.first` in two cases argues for your way of doing things.